pkg/component/istio/monitoring.go

```diff
@@ -80,125 +82,94 @@ const (
+					monitoringMetricPilotConflictOutboundListenerTcpOverCurrentTcp,
+					monitoringMetricPilotK8sCfgEvents,
+					monitoringMetricPilotProxy_convergenceTimeBucket,
+					monitoringMetricPilotServices,
+					monitoringMetricPilotTotalXdsInternalErrors,
+					monitoringMetricPilotTotalXdsRejects,
+					monitoringMetricPilotVirtServices,
+					monitoringMetricPilotXds,
+					monitoringMetricPilotXdsCdsReject,
+					monitoringMetricPilotXdsEdsReject,
+					monitoringMetricPilotXdsLdsReject,
+					monitoringMetricPilotXdsPushContextErrors,
+					monitoringMetricPilotXdsPushes,
+					monitoringMetricPilotXdsRdsReject,
+					monitoringMetricPilotXdsWriteTimeout,
+					monitoringMetricProcessCpuSecondsTotal,
+					monitoringMetricProcessOpenFds,
+					monitoringMetricProcessResidentMemoryBytes,
+					monitoringMetricProcessVirtualMemoryBytes,
+				),
+			}},
+		},
 	}
 
-	monitoringAllowedMetricsIstioIngressGateway = []string{
-		monitoringMetricEnvoyClusterUpstreamCxActive,
-		monitoringMetricEnvoyClusterUpstreamCxConnectFail,
-		monitoringMetricEnvoyClusterUpstreamCxTotal,
-		monitoringMetricEnvoyClusterUpstreamCxTxBytesTotal,
-		monitoringMetricEnvoyServerHotRestartEpoch,
-		monitoringMetricGoGoroutines,
-		monitoringMetricGoMemstatsAllocBytes,
-		monitoringMetricGoMemstatsHeapAllocBytes,
-		monitoringMetricGoMemstatsHeapInuseBytes,
-		monitoringMetricGoMemstatsHeapSysBytes,
-		monitoringMetricGoMemstatsStackInuseBytes,
-		monitoringMetricIstioBuild,
-		monitoringMetricIstioRequestBytesBucket,
-		monitoringMetricIstioRequestBytesSum,
-		monitoringMetricIstioRequestDurationMillisecondsBucket,
-		monitoringMetricIstioRequestDurationSecondsBucket,
-		monitoringMetricIstioRequestsTotal,
-		monitoringMetricIstioResponseBytesBucket,
-		monitoringMetricIstioResponseBytesSum,
-		monitoringMetricIstioTcpConnectionsClosedTotal,
-		monitoringMetricIstioTcpConnectionsOpenedTotal,
-		monitoringMetricIstioTcpReceivedBytesTotal,
-		monitoringMetricIstioTcpSentBytesTotal,
-		monitoringMetricProcessCpuSecondsTotal,
-		monitoringMetricProcessOpenFds,
-		monitoringMetricProcessResidentMemoryBytes,
-		monitoringMetricProcessVirtualMemoryBytes,
+	serviceMonitorIstioIngress = &monitoringv1.ServiceMonitor{
+		ObjectMeta: monitoringutils.ConfigObjectMeta(monitoringPrometheusJobNameIstioIngressGateway, v1beta1constants.IstioSystemNamespace, aggregate.Label),
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector:          metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.LabelApp: "istio-ingressgateway"}},
+			NamespaceSelector: monitoringv1.NamespaceSelector{Any: true},
+			Endpoints: []monitoringv1.Endpoint{{
+				Path: "/stats/prometheus",
+				Port: "tls-tunnel",
+				RelabelConfigs: []*monitoringv1.RelabelConfig{{
+					SourceLabels: []monitoringv1.LabelName{"__meta_kubernetes_pod_ip"},
+					Action:       "replace",
+					TargetLabel:  "__address__",
+					Regex:        `(.+)`,
+					Replacement:  "${1}:15020",
+				}},
```

Comment:
Is this really doing the same thing as before? If so, that is great as it is much more concise and understandable.

Reply:
Same as above. Great simplification :-) .